frontend/src/HomePage.jsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react";
+import { useRef } from "react";
+import "./homepage.css";
+import { boards } from "./data";
+import { Link } from "react-router-dom";
+
+import Header from "./components/Header/Header";
+import SearchBar from "./components/SearchBar/SearchBar";
+import FilterButtons from "./components/FilterButtons/FilterButtons";
+import BoardList from "./components/BoardList/BoardList";
+import Footer from "./components/Footer/Footer";
+import NewBoardForm from "./components/NewBoardForm/NewBoardForm";
+
+export default function HomePage() {
+  const [newBrdFormOpened, setNewBrdFormOpened] = useState(false);
+  const [boardList, setBoardList] = useState(boards);
+
+  function handleSearch(searchInput) {}
+
+  function handleClear() {}
+
+  //this function will be passed down to FilterButtons component so that in that component we can determine which filter user clicks and call this function and send result back up to parent in the form of "filterType". All filtering logic and updating of boardList happens here so we dont have to pass all that down
+  function handleFilter(filterType) {
+    if (filterType === "all") {
+      setBoardList(boards);
+    } else if (filterType === "recent") {
+      //sort the array by recent date and then splice it so only 6 get displayed
+    } else if (filterType === "celebration") {
+      const filteredList = boardList.filter(
+        (board) => board.category === "celebration"
+      );
+      setBoardList(filteredList);
+    } else if (filterType === "thank you") {
+      const filteredList = boards.filter(
+        (board) => board.category === "thank you"
+      );
+      console.log(filteredList);
+      setBoardList(filteredList);
+    } else {
+      const filteredList = boards.filter(
+        (board) => board.category === "inspiration"
+      );
+      setBoardList(filteredList);
+    }
```

Comment:
When you have multiple if-statements, I prefer to use a switch-case as I feel is more readable, and sometimes easy to put the same logic together:
```suggestion
    switch(filterType) {
      case "all":
        setBoardList(boards);
        break;
      case "recent":
        // sort the array by recent date and then splice it so only 6 get displayed
        break;
      case "celebration":
      case "thank you":
      case "inspiration":
        const filteredList = boards.filter(
          (board) => board.category === filterType
        );
        setBoardList(filteredList);
        break;
      default:
        // maybe log this or treat as the "all"
    }
```

Reply:
Honestly you are totally right. This code is way more readable I dont know why I didnt think about using a switch case. I will implement it this way instead.